service/gcs/core/gcs/storage.go

```diff
@@ -331,7 +330,7 @@ func (c *gcsCore) writeConfigFile(id string, config oci.Spec) error {
 	}
 	defer configFile.Close()
 	writer := bufio.NewWriter(configFile)
-	if err := json.NewEncoder(writer).Encode(config); err != nil {
+	if err := utils.EncodeJSONWithHresult(writer, config); err != nil {
```

Comment:
That's fair. It makes sense to reserve this HRESULT for errors that are specifically caused by invalid JSON from the host side. I can remove everything that isn't caused by errors parsing host-originating JSON.